src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -375,4 +380,79 @@ absl::StatusOr<std::string> AkidFromCrl(X509_CRL* crl) {
   return ret;
 }
 
+bool IsPemCertificateChainNonEmptyAndValid(absl::string_view cert_chain_pem) {
```

Comment:
It's pretty unclear what "valid" means here. "Valid" means a lot of things in X.509 from:

* The surface-level parse (without reaching into the extensions) succeeds
* Extensions that we know about can be parsed
* The certificate chains up to one of some set of trust anchors
* Variations on the above due to implementation issues (it is _not_ guaranteed that, if `d2i_X509` accepts an input in one implementation, that it will also be accepted in another. There's too much tech debt in the X.509 ecosystem for that)

Reply:
Based on @bob-beck's suggestion, I changed the API signature so now it parses and returns the vector of `X509`s. Also changed the name to simply `ParsePemCertificateChain` to hopefully make it more clear.